cmd/srcd/cmd/sql_test.go

```diff
@@ -211,6 +214,97 @@ func (s *SQLTestSuite) TestREPL() {
 	require.Contains(out.String(), expected)
 }
 
+func (s *SQLTestSuite) TestIndexesWorkdirChange() {
+	require := s.Require()
+
+	// use engine repo itself to avoid cloning anything
+	wd, err := os.Getwd()
+	require.NoError(err)
+	enginePath := path.Join(wd, "..", "..", "..")
+
+	// workdir 1
+	_, err = s.RunInit(context.TODO(), enginePath)
+	require.NoError(err)
+
+	buf, err := s.RunSQL(context.TODO(), "CREATE INDEX repo_idx ON repositories USING pilosa (repository_id)")
+	require.NoError(err, buf.String())
+
+	time.Sleep(1 * time.Second) // wait for index to be built
+
+	s.testQueryWithIndex(require)
+
+	// workdir 2
+	repoPath := filepath.Join(s.testDir, "reponame")
+	err = os.Mkdir(repoPath, os.ModePerm)
+	require.NoError(err)
+
+	cmd := exec.Command("git", "init", repoPath)
+	err = cmd.Run()
+	require.NoError(err)
+
+	_, err = s.RunInit(context.TODO(), s.testDir)
+	require.NoError(err)
+
+	buf, err = s.RunSQL(context.TODO(), "SHOW INDEX FROM repositories")
```

Comment:
We could reuse the `testQueryWithIndex` by changing its signature. Adding the `repo_id string` and `contains bool` should be enough to cover this case.

Reply:
I'm not sure if it makes the code simpler. But applied.